dbt_coves/tasks/blue_green/clone_db.py

```diff
@@ -0,0 +1,215 @@
+                sql = f"create schema {green_database}.{schema['name']} clone {blue_database}.{schema['name']};"
+                threaded_run_commands.register_command(sql)
+        threaded_run_commands.run()
+        console.print(f"Cloned schemas in {time.time() - self.time_check} seconds.")
+        self.time_check = time.time()
+        # Copy grants from Blue DB schemas
+        console.print(
+            f"Cloning [u]schema grants[/u] from [blue]{self.blue_database}[/blue] to"
+            f"[green]{self.green_database}[/green]"
+        )
+        for schema in schemas:
+            if schema["name"] not in self._list_of_schemas_to_exclude:
+                grants_sql_stg_1 = f"""show grants on schema {blue_database}.{schema['name']}"""
+                dict_cursor.execute(grants_sql_stg_1)
+                grants = dict_cursor.fetchall()
+                for grant in grants:
+                    sql = (
+                        f"GRANT {grant['privilege']} ON {grant['granted_on']} {green_database}.{schema['name']}"
+                        f"TO ROLE {grant['grantee_name']};"
+                    )
+                    # Load SQL into the threaded commands to run.
+                    threaded_run_commands.register_command(sql)
+        threaded_run_commands.run()
+        print(f"Cloned grants to schemas in {time.time() - self.time_check} seconds.")
+        self.time_check = time.time()
+
+
+class ThreadedRunCommands:
+    """Helper class for running queries across a configurable number of threads"""
+
+    def __init__(self, con, threads):
+        self.threads = threads
+        self.register_command_thread = 0
+        self.thread_commands = [[] for _ in range(self.threads)]
+        self.con = con
+
+    def register_command(self, command: str):
+        """
+        Register a sql command to be run in a thread.
+
+        Args:
+            command: A SQL string to be run.
+
+        Returns:
+            None
+        """
+        self.thread_commands[self.register_command_thread].append(command)
+        if self.register_command_thread + 1 == self.threads:
+            self.register_command_thread = 0
+        else:
+            self.register_command_thread += 1
+
+    def run_commands(self, commands):
+        """
+        Loops over the commands passing off to the "run
+        Args:
+            commands: A list of SQL commands to be run async.
+
+        Returns:
+            None
+        """
+        for command in commands:
+            self.con.cursor().execute_async(command)
+
+    def run(self):
+        """
+        Run the commands in the threads.
+
+        Returns:
+            None
+        """
+        procs = []
+        for v in self.thread_commands:
+            proc = threading.Thread(target=self.run_commands, args=(v,))
+            procs.append(proc)
+            proc.start()
+        # complete the processes
+        for proc in procs:
+            proc.join()
+
+
+# if __name__ == "__main__":
```

Comment:
Could we avoid keeping this commented out and test this differently?

Reply:
@ssassi I agree, it belongs to the code Coldbore sent us, and its testing case. I'd like to discuss how all this should be tested.